compiler/testData/diagnostics/tests/extensions/contextReceivers/functionalType.fir.kt

```diff
@@ -0,0 +1,39 @@
+class Param
+class C
+class R
+
+context(C)
+fun R.f1(g: context(C) R.(Param) -> Unit) {
+    g(this@C, this@R, Param())
+}
+
+context(C)
+fun f2(g: context(C) (Param) -> Unit) {
+    g(this@C, Param())
+}
+
+context(C)
+fun R.f3(g: context(C) R.() -> Unit) {
+    g(this@C, this@R)
+}
+
+context(C)
+fun f4(g: context(C) () -> Unit) {
+    g(this@C)
+}
+
+fun test() {
+    val lf1: context(C) R.(Param) -> Unit = { _ -> }
+    val lf2: context(C) (Param) -> Unit = { _ -> }
+    val lf3: context(C) R.() -> Unit = { }
```

Comment:
Also, it's worth having a BlackBox for such cases

Reply:
Added blackbox, but it still fails for the `invoke` call on `FunctionN` because its declaration has the wrong parameter order (extension receiver is on the first place). Trying to figure out where it is generated on BE.